server/src/server.common.ts

```diff
@@ -202,25 +224,52 @@ export function startServer(connection: _Connection) {
     return [result.contents, result.uri];
   }
 
-  // Attempt to get contents of declared file for a given uri
-  async function getFileContents(originalUri: URI, fileUri: string): Promise<{ contents: string; uri: URI }> {
-    const uri = Utils.resolvePath(originalUri, "..", fileUri);
-
-    let contents = documents.get(uri.toString())?.getText() as string;
-    if (!contents) {
-      contents = await connection.sendRequest("getFileContents", uri.toString());
+  // Parse and validate DSL, whether it is a regular or modular model file
+  async function getDiagnosticsForDsl(dsl: string, uri?: string): Promise<Diagnostic[]> {
+    try {
+      const transform = transformer.transformDSLToJSONObject(dsl);
+      if (transform.schema_version) {
+        // If regular module
+        validator.validateDSL(dsl);
+      } else if (uri) {
+        // If a modular model & has a URI, validate against the fga.mod file
+        // Get the closest fga.mod files
+        const fgaModFile = await clientRequests.getFileUp(URI.parse(uri), "fga.mod");
+        if (fgaModFile?.contents && fgaModFile.uri) {
+          // Get path from URI to match diagnostics
+          const modFilePath = fgaModFile.uri.path;
+          const filePath = URI.parse(uri).path.replace(modFilePath.substring(0, modFilePath.lastIndexOf("/") + 1), "");
+          // Return only diagnostics that match the models file path
+          return (await validateFgaMod(fgaModFile.contents, modFilePath)).diagnostics.filter((d) => {
+            return d.data.file === filePath;
+          });
+        }
+      }
+    } catch (err) {
+      if (
+        err instanceof errors.DSLSyntaxError ||
```

Comment:
Hmm.. Maybe we should have all these errors extend the same base type in language so that you can have oly one check here.

cc @ewanharris

Reply:
Yeah that probably would make this nicer, maybe just an emptY `LanguageError` class that they all extend?

Reply:
Will be resolved by https://github.com/openfga/language/pull/213